src/main/java/spark/Request.java

```diff
@@ -513,37 +513,69 @@ public String protocol() {
         return Collections.unmodifiableMap(params);
     }
 
-    private static List<String> getSplat(List<String> request, List<String> matched) {
-        int nbrOfRequestParts = request.size();
-        int nbrOfMatchedParts = matched.size();
+    static List<String> getSplat(List<String> request, List<String> matched) {
+        List<String> splat = getSplat(matched, request, 0, 0, new ArrayList<>());
 
-        boolean sameLength = (nbrOfRequestParts == nbrOfMatchedParts);
+        return Collections.unmodifiableList(splat);
+    }
 
-        List<String> splat = new ArrayList<>();
+    private static List<String> getSplat(List<String> matched, List<String> request, int i, int j, List<String> splat) {
```

Comment:
here's my suggestion, a bit shorter and with only small differences from RouteEntry method. In the end, result should be reversed. Btw. it's possible I made a couple of off by one errors with indices :)
```
    private boolean matchPath(List<String> matched, List<String> request, int i, int j, List<String> result) {
        if (i == request.size() && j == matched.size()) return true;
        if (i == request.size() || j == matched.size()) return false;

        String requestPart = request.get(i);
        String matchedPart = matched.get(j);

        if (SparkUtils.isSplat(matchedPart)) {
            if (j == matched.size() - 1) {
                result.add(createSplat(request, i, request.size()));
                return true;
            }
            for (int k = i + 1; k < requestPart.length(); k++) {
                if (matchPath(request, matched, k, j + 1, result)) {
                    result.add(createSplat(request, i, k));
                    return true;
                }
            }
            return false;
        } else if (SparkUtils.isParam(matchedPath) || matchedPath.equals(requestPath)) {
            return matchPath(request, matched, i + 1, j + 1, result);
        } else {
            return false;
        }
    }

    private String createSplat(List<String> request, int start, int end) {
        try {
            String splat = request.stream().limit(end).skip(start).collect(Collectors.joining("/"));
            return URLDecoder.decode(splat, "UTF-8");
        } catch (UnsupportedEncodingException e) {
            return "";
        }
    }
```

Reply:
i ended up using it, thank you for the suggestion!